### DISCUSSION TITLE
AST of Odin function

### DISCUSSION THREAD

**Discussion by ib00** [odin-lang/Odin#2067] [Q&A] [answered]
Is there a way to dump (or traverse) AST of a function?

I am looking into Odin for using it to create differentiable programs (by creating an adjoint of it).

**Comment by Kelimion** [comment, marked as answer]
There is the `core:odin` package which will let you parse a package, file or a string of Odin code.

I suggest visiting `#ols` on the [Odin Discord](https://discord.com/invite/sVBPHEv) if you get stuck. The Odin Language Server (OLS) heavily uses this package.
